### ISSUE TITLE
regression: unused imports false positive on nightly for the mach crate

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust#55411]
With the latest version of mach, running `cargo test` shows:

```shell
warning: unused import: `vm_region::*`                                                                                                                    
   --> src/vm.rs:200:9                                                                                                                                    
    |                                                                                                                                                     
200 |     use vm_region::*;                                                                                                                               
    |         ^^^^^^^^^^^^                                                                                                                                
    |                                                                                                                                                     
    = note: #[warn(unused_imports)] on by default                                                                                                         
                                                                                                                                                          
warning: unused import: `vm_types::*`                                                                                                                     
   --> src/vm.rs:202:9                                                                                                                                    
    |                                                                                                                                                     
202 |     use vm_types::*;                                                                                                                                
    |         ^^^^^^^^^^^     
```

This warning did not used to trigger, and is incorrect (removing the import breaks the build).
